clients/drcachesim/tools/invariant_checker.h

```diff
@@ -140,14 +140,15 @@ class invariant_checker_t : public analysis_tool_t {
             bool is_predicated = false;
             uint num_memory_read_access = 0;
             uint num_memory_write_access = 0;
-            addr_t branch_target;
+            addr_t branch_target = 0;
         };
         struct instr_info_t {
-            memref_t memref;
-            decoding_info_t decoding;
+            memref_t memref = {};
+            decoding_info_t decoding = {};
```

Comment:
But you have default values for all the fields so why set those and then do a separate memset to zero?

Reply:
Yes, you need the memref memset: that is what I pointed out should be added in the prior review.  It is a C struct.  What you don't need is the decoding_info_t which is a C++ struct with every field initialized in its default constructor -- that is why I have the comment on this line, not the line above it.